enclave-runtime/src/tls_ra.rs

```diff
@@ -24,11 +24,11 @@ use webpki::DNSName;
 struct ClientAuth<A> {
 	outdated_ok: bool,
 	skip_ra: bool,
-	attestation_ocall: Arc<A>,
+	attestation_ocall: A,
```

Comment:
This is okay as long as we don't have state in the OCallApi (which we don't at the moment, afaik)

Reply:
I am still also trying to exactly understand, when to use arcs. But I think you only need it for shared mutability. I believe, it is still fine if we have state, as long we don't need `&mut` simultaneously, or if we need to do expensive clones.

Reply:
I'd say shared state, not shared mutability - since a `T` in `Arc<T>` cannot be used as `mut`, it's explicitely forbidden, you have to use locks, like `RwLock` in order to have `Arc<T>` with mutable state.

Reply:
Hmm, ok I see. To be more precise, it would probably be shared ownership. Shared state by itself should not be a problem.

Reply:
yes, shared ownership is correct. And usually you want that in cases where you have stateful objects.